node/txnSyncConn.go

```diff
@@ -141,11 +144,12 @@ func (tsnc *transcationSyncNodeConnector) SendPeerMessage(netPeer interface{}, m
 		return
 	}
 
-	if err := unicastPeer.Unicast(context.Background(), msg, protocol.Txn2Tag, func(enqueued bool, sequenceNumber uint64) {
+	if err := unicastPeer.Unicast(context.Background(), msg, protocol.Txn2Tag, func(enqueued bool, sequenceNumber uint64) error {
 		if callbackErr := callback(enqueued, sequenceNumber); callbackErr != nil {
-			// disconnect from peer - the transaction sync wasn't able to process message sending confirmation
-			tsnc.node.net.Disconnect(unicastPeer)
+			// disconnect from peer - return a non-nil error
+			return errDisconnectFromNetworkPeer
```

Comment:
why not return callbackErr ?

Reply:
I guess that you're correct here. In this case, we can just return the error. I'll make the change.